File: OCH_Tao/ABC/B/300.py
import numpy as np
H, W = map(int, input().split())
A = []
for i in range(H):
  A.append(list(input()))
A = np.array(A)
B = []
for i in range(H):
  B.append(list(input()))
B = np.array(B)
for i in np.ndindex(H, W):
  if np.all(np.roll(A, i, axis=(0, 1)) == B):
    print("Yes")
    break
else:
  print("No")

File: OCH_Tao/ABC/B/300.js
const input = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n");
const [H, W] = input.shift().split(" ").map(x => parseInt(x, 10));
const A = input.slice(0, H);
const B = input.slice(H);
for (let i = 0; i < H; i++) {
  for (let j = 0; j < W; j++) {
    check: {
      for (let x = 0; x < H; x++) {
        for (let y = 0; y < W; y++) {
          if (A[(x + i) % H][(y + j) % W] !== B[x][y]) {
            break check;
          }
        }
      }
      console.log("Yes");
      process.exit();
    }
  }
}
console.log("No");